yarn-project/noir-contracts/src/contracts/token_bridge_contract/src/main.nr

```diff
@@ -0,0 +1,169 @@
+mod util;
+mod token_interface;
+
+// Minimal implementation of the token bridge that can move funds between L1 <> L2.
+// The bridge has a corresponding Portal contract on L1 that it is attached to 
+// And corresponds to a Token on L2 that uses the `AuthWit` accounts pattern.
+// Bridge has to be set as a minter on the token before it can be sued
+
+contract TokenBridge {
+    use dep::aztec::{
+        context::{Context},
+        state_vars::{public_state::PublicState},
+        types::type_serialisation::field_serialisation::{
+            FieldSerialisationMethods, FIELD_SERIALISED_LEN,
+        },
+        types::address::{AztecAddress, EthereumAddress},
+        oracle::compute_selector::compute_selector,
+    };
+
+    use crate::token_interface::Token;
+    use crate::util::{get_mint_content_hash, get_withdraw_content_hash, compute_secret_hash};
+
+    // Storage structure, containing all storage, and specifying what slots they use.
+    struct Storage {
+        token: PublicState<Field, 1>,
+    }
+
+    impl Storage {
+        fn init(context: Context) -> pub Self {
+            Storage {
+                token: PublicState::new(
+                    context,
+                    1,
+                    FieldSerialisationMethods,
+                ),
+            }
+        }
+    }
+
+    // Constructs the contract.
+    #[aztec(private)]
+    fn constructor() {
+        // Currently not possible to execute public calls from constructor as code not yet available to sequencer.
+        // let selector = compute_selector("_initialize((Field))");
+        // let _callStackItem = context.call_public_function(context.this_address(), selector, [context.msg_sender()]);
+    }
+
+    // Consumes a L1->L2 message and calls the token contract to mint the appropriate amount publicly
+    #[aztec(public)]
+    fn deposit_public(
+        amount: Field,
+        msg_key: Field,
+        secret: Field,
+        canceller: EthereumAddress,
+    ) -> Field {
+        let storage = Storage::init(Context::public(&mut context));
+
+        let content_hash = get_mint_content_hash(amount, context.msg_sender(), canceller.address);
```

Comment:
There should be two different content_hashes, one for public and one for private. Say you do:

```solidity
bytes32 content = sha256ToField(
    abi.encodeWithSelector(
        "mint_public(bytes32,uint256,address)", 
        to, 
        amount, 
        canceller
);
```

And then a separate 
```solidity
bytes32 content = sha256ToField(
    abi.encodeWithSelector(
        "mint_private(uint256,bytes32,address)", 
        amount,
        secretHash, 
        canceller
);
```

With these two, you will be unambigious on what you are calling. Also, it is easier for you to let someone else perform the deposit action on L2 on your behalf without, in private use the `to` that was part of the content, and for public use the `secretHash` for the secretHash that is used when minting the asset.

Reply:
Note that this require two flows on the L1 contract as well. 

For withdrawals, you can get away with just having one content hash function, as you cannot consume it in private or public when on L1, so no need for both options.